src/query/update.rs

```diff
@@ -66,6 +67,62 @@ impl UpdateStatement {
         self
     }
 
+    /// From table.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{tests_cfg::*, *};
+    ///
+    /// let query = Query::update()
+    ///     .table(Glyph::Table)
+    ///     .value(
+    ///         Glyph::Tokens,
+    ///         SimpleExpr::Column(ColumnRef::TableColumn(
+    ///             SeaRc::new(Char::Table),
+    ///             SeaRc::new(Char::Character),
+    ///         )),
+    ///     )
+    ///     .from(Char::Table)
+    ///     .cond_where(
+    ///         SimpleExpr::Column(ColumnRef::TableColumn(
+    ///             SeaRc::new(Glyph::Table),
+    ///             SeaRc::new(Glyph::Image),
+    ///         ))
+    ///         .eq(SimpleExpr::Column(ColumnRef::TableColumn(
+    ///             SeaRc::new(Char::Table),
+    ///             SeaRc::new(Char::UserData),
+    ///         ))),
+    ///     )
+    ///     .to_owned();
+    ///
+    ///
+    /// assert_eq!(
+    ///     query.to_string(MysqlQueryBuilder),
+    ///     "UPDATE `glyph` SET `tokens` = `character`.`character` FROM `character` WHERE `glyph`.`image` = `character`.`user_data`"
```

Comment:
The syntax of MySQL appears similar, but not identical. Here a MySQL UPDATE JOIN could be used to achieve the same:

```sql
UPDATE `glyph`
JOIN `character`
  ON `glyph`.`image` = `character`.`user_data`
SET `glyph`.`tokens` = `character`.`character`;
```

Reply:
@gronke Took a wild swing at at least getting started here, and extracted the specific behavior into a discrete method (`prepare_update_from_table_refs`) so it can be overridden as needed by whichever dialect. I've got the MySQL-specific handling as a `unimplemented!` call at the moment though, cause I've got a case of end-of-year-holidays-brain.

If you have a second, would you take a look at the changes and let me know if there's a better / more preferable way of doing this please let me know? Otherwise I'll carry on as time allows 😁